sormas-backend/src/main/java/de/symeda/sormas/backend/externalemail/AttachmentService.java

```diff
@@ -0,0 +1,204 @@
+                } else {
+                    PdfConverter converter = getConverter(fileExtension);
+                    fileName = converter.getConvertedFileName(fileName);
+                    File converted = converter.convert(document);
+                    encryptedPdf = encryptPdf(converted, passowrd);
+                    converted.delete();
+                }
+
+                return new AbstractMap.SimpleEntry<>(encryptedPdf, fileName);
+            } catch (IOException e) {
+                // not really expected to happen
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private File encryptPdf(File pdf, String password) throws IOException {
+        PDDocument pdd = Loader.loadPDF(pdf);
+
+        AccessPermission ap = new AccessPermission();
+        StandardProtectionPolicy stpp = new StandardProtectionPolicy(password, password, ap);
+        stpp.setEncryptionKeyLength(128);
+        stpp.setPermissions(ap);
+        pdd.protect(stpp);
+
+        String encryptedFile = getTmpFilePathForConversion() + "_encrypted" + PDF_FILE_EXTENSION;
+        pdd.save(encryptedFile);
+        pdd.close();
+
+        return new File(encryptedFile);
+    }
+
+    private PdfConverter getConverter(String fileExtension) {
+        Optional<String> converterKey = converters.keySet().stream().filter(k -> k.contains(fileExtension)).findFirst();
+        if (converterKey.isEmpty()) {
+            throw new IllegalArgumentException("No converter found for file extension " + fileExtension);
+        }
+
+        return converters.get(converterKey.get());
+    }
+
+    private Path getTmpFilePathForConversion() {
+        Path path = Paths.get(configFacade.getTempFilesPath());
+        String fileName = TEMP_FILE_PREFIX + "_converted_" + DateHelper.formatDateForExport(new Date()) + "_" + RANDOM.nextInt(Integer.MAX_VALUE);
+
+        return path.resolve(fileName);
+
+    }
+
+    private interface PdfConverter {
+
+        File convert(File file) throws IOException;
+
+        default String getConvertedFileName(String fileName) {
+            return fileName.substring(0, fileName.lastIndexOf(".")) + PDF_FILE_EXTENSION;
+        }
+    }
+
+    private class DocXConverter implements PdfConverter {
+
+        @Override
+        public File convert(File file) throws IOException {
+            String convertedFilePath = getTmpFilePathForConversion() + PDF_FILE_EXTENSION;
+            try (InputStream inputStream = new FileInputStream(file); OutputStream outputStream = new FileOutputStream(convertedFilePath)) {
+                XWPFDocument document = new XWPFDocument(inputStream);
+                PdfOptions options = PdfOptions.create();
+                // Convert .docx file to .pdf file
+                fr.opensagres.poi.xwpf.converter.pdf.PdfConverter.getInstance().convert(document, outputStream, options);
```

Comment:
The convert method can throw XWPFConverterException, IOException exceptions too

Reply:
It is a runtime exception, will be caught and logged by a generic error handler. If it somehow happens it can be considered a system issue so it's ok that the user will see a generic server error message.